test/conftest.py

```diff
@@ -21,9 +34,53 @@ def app() -> Flask:
                 'client_id': 'server_client_id',
                 'client_secret': 'server_client_secret',
             }
-        }
+        },
     })
 
+    ctx = app.app_context()
+    ctx.push()
+
+    def teardown():
+        ctx.pop()
+
+    request.addfinalizer(teardown)
+    return app
+
+
+@fixture(scope='session', autouse=True)
+def database(app: Flask, request: FixtureRequest) -> SQLAlchemy:
+    if os.path.exists(TEST_DATABASE_PATH):
+        os.unlink(TEST_DATABASE_PATH)
+
+    db.app = app
+    db.create_all()
+
+    def teardown():
+        db.drop_all()
+        os.unlink(TEST_DATABASE_PATH)
+
+    request.addfinalizer(teardown)
+    return db
+
+
+@fixture(scope='function', autouse=True)
+def session(database: SQLAlchemy, request: FixtureRequest) -> scoped_session:
+    connection = database.engine.connect()
+    transaction = connection.begin()
+
+    options = dict(bind=connection, binds={})
+    session = database.create_scoped_session(options=options)
+
+    database.session = session
+
```

Comment:
seems to be http://xunitpatterns.com/Transaction%20Rollback%20Teardown.html so I'd link it here